### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by AhmadReza2003H**

### CODE REVIEW

File: Date.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    } else{
+        return Date(day , mounth);
+    }
+}
+Date Date::plusDay(int day) const{
+    if(day > Date::remainDays || day < 0){
+        throw "Invalid Argument";
+    } else {
+        int newDay = Date::day;
+        int newMounth = Date::mounth;
```
Comment (mirmilad) [inline Date.cpp]:
```suggestion
        int newDay = this->day;
        int newMounth = this->mounth;

```

File: Date.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+}
+Date Date::setDate(int day , int mounth) const{
+    if(mounth > 12 || mounth < 1 || mounthDay.at(mounth) < day){
+        throw "Invalid Argument";
+    } else{
+        return Date(day , mounth);
+    }
+}
+Date Date::plusDay(int day) const{
```
Comment (mirmilad) [inline Date.cpp]: I would implement `plusDay` and `minusDay` in another way. You already of the `dayOfYear`. You can simply add or subtract the `day` from `dayOfYear` and then have a function to calculate `day` and `month` out of `dayOfYear`.

File: Date.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                newMounth--;
+            } else {
+                break;
+            }
+        }
+        return Date(newDay , newMounth);
+    }
+}
+void Date::print() const{
+    printf("\n%s , %s %d 2013\n", dayName.at(Date::dayOfYear%7).c_str() , mounthName.at(Date::mounth).c_str() , Date::day);
```
Comment (mirmilad) [inline Date.cpp]: Why do you need `\n` in the beginning? 
It's C++, use `cout` instead.

File: maps.h [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#ifndef _DAY_H
```
Comment (mirmilad) [inline maps.h]: Usually, you should use the file path to create an import guard, otherwise, it's hard to maintain.

File: maps.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#include <unordered_map>
+#include <string>
+#include "maps.h"
+using namespace std;
+unordered_map<int, int> mounthDay = {
```
Comment (mirmilad) [inline maps.cpp]: It must be const, otherwise, it can be changed at runtime and make your calendar wrong.

File: Date.cpp [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+Date Date::minusDay(int day) const {
+    if(day > this->dayOfYear || day < 0){
+        throw "Invalid Argument";
+    } else {
+        int newDay = this->dayOfYear - day;
+        return Date(newDay);
+    }
+}
```
Comment (mirmilad) [inline Date.cpp]: These two functions still have the same logic, you can put the logic in a private function and call it in these functions.

File: maps.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#include <unordered_map>
+#include <string>
+#include "maps.h"
+using namespace std;
+const unordered_map<int, int> mounthDay = {
```
Comment (mirmilad) [inline maps.cpp:5]: Are you sure `const` is enough and no body can change the content of this map at runtime?

Comment (AhmadReza2003H) [reply maps.cpp:5]: yes it's enough

**Review by mirmilad — APPROVED** [review]

**Pull request merged**